new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTraefikLoadBalancer.java

```diff
@@ -0,0 +1,304 @@
+          replicaCount,
+          IMAGE);
+
+      // create model in image domain
+      logger.info("Creating model in image domain {0} in namespace {1} using docker image {2}",
+          domainUid, domainNamespace, MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG);
+      createDomainAndVerify(domain, domainNamespace);
+
+      logger.info("Check admin service {0} is created in namespace {1}",
+          adminServerPodName, domainNamespace);
+      checkServiceExists(adminServerPodName, domainNamespace);
+
+      // check admin server pod is ready
+      logger.info("Waiting for admin server pod {0} to be ready in namespace {1}",
+          adminServerPodName, domainNamespace);
+      checkPodReady(adminServerPodName, domainUid, domainNamespace);
+
+      // check managed server services created
+      for (int i = 1; i <= replicaCount; i++) {
+        logger.info("Check managed server service {0} is created in namespace {1}",
+            managedServerPrefix + i, domainNamespace);
+        checkServiceExists(managedServerPrefix + i, domainNamespace);
+      }
+
+      // check managed server pods are ready
+      for (int i = 1; i <= replicaCount; i++) {
+        logger.info("Wait for managed server pod {0} to be ready in namespace {1}",
+            managedServerPrefix + i, domainNamespace);
+        checkPodReady(managedServerPrefix + i, domainUid, domainNamespace);
+      }
+
+      //create ingress resource - rules for loadbalancing
+      Map<String, Integer> clusterNameMsPortMap = new HashMap<>();
+      clusterNameMsPortMap.put("cluster-1", 8001);
+      logger.info("Creating ingress resource for domain {0} in namespace {1}", domainUid, domainNamespace);
+      createTraefikIngressForDomainAndVerify(domainUid, domainNamespace, 0, clusterNameMsPortMap, true);
+
+      logger.info("Getting node port for admin server default channel");
+      int serviceNodePort = assertDoesNotThrow(()
+          -> getServiceNodePort(domainNamespace, adminServerPodName + "-external", "default"),
+          "Getting admin server node port failed");
+
+      logger.info("Deploying application {0} in domain {1} cluster target cluster-1",
+          clusterViewAppPath, domainUid);
+      ExecResult result = DeployUtil.deployUsingRest(K8S_NODEPORT_HOST,
+          String.valueOf(serviceNodePort),
+          ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT,
+          "cluster-1", clusterViewAppPath, null, domainUid + "clusterview");
+      assertNotNull(result, "Application deployment failed");
+      logger.info("Application deployment returned {0}", result.toString());
+      assertEquals("202", result.stdout(), "Deployment didn't return HTTP status code 202");
+
+      verifyLoadbalancing(domainUid, replicaCount, managedServerNameBase);
+    }
+
+    // verify load balancing works when 2 domains are running in the same namespace
+    for (String domainUid : domains) {
+      verifyLoadbalancing(domainUid, replicaCount, managedServerNameBase);
```

Comment:
Does the clusterView Servlet assures that the response is coming form domain1/domain2. Does it display the unique managed server name from each domain ?

Reply:
Yes, the domain names are bound in JNDI trees of the managed servers and later verified while accessing a particular managed server response.